libs/design/src/core/sticky/sticky-tracker.directive.ts

```diff
@@ -0,0 +1,216 @@
+import { DOCUMENT } from '@angular/common';
+import {
+  Directive,
+  ElementRef,
+  NgZone,
+  OnDestroy,
+  AfterViewInit,
+  Renderer2,
+  Inject,
+} from '@angular/core';
+
+interface ComputedStylesData {
+  stickyElementStyle: CSSStyleDeclaration;
+  parentElements: HTMLElement[];
+  parentStyles: CSSStyleDeclaration[];
+}
+
+@Directive({
+  selector: '[daffStickyTracker]',
+})
+export class DaffStickyTrackerDirective implements AfterViewInit, OnDestroy {
+  private sentinelObserver?: IntersectionObserver;
+  private readonly className = 'is-pinned';
+  private lastPinnedState: boolean | null = null;
+  private debounceTimeout?: number;
+  private isBottomSticky = false;
+  private stickyPosition = '0';
+  private sentinelElement?: HTMLElement;
+
+  constructor(
+    private readonly elementRef: ElementRef<HTMLElement>,
+    private readonly renderer: Renderer2,
+    private readonly ngZone: NgZone,
+    @Inject(DOCUMENT) private readonly document: Document,
+  ) {}
+
+  /**
+   * @docs-private
+   */
+  ngAfterViewInit(): void {
+    this.ngZone.runOutsideAngular(() => {
```

Comment:
I think, instead of doing this outside the zone, we should do this with a `afterRender` callback. Can you do a performance profile of this before/after this change?